lib/src/entrypoint.dart

```diff
@@ -642,4 +646,127 @@ class Entrypoint {
+    // Count how many scripts we could potentially run.
+    var allScripts = <String, List<String>>{};
+
+    for (var package in result.changedPackages) {
+      var scripts = result.pubspecs[package].afterInstall;
+      if (scripts.isNotEmpty) allScripts[package] = scripts;
+    }
+
+    // If no scripts need to be run, don't bother updating the cache.
+    if (allScripts.isEmpty) return;
+
+    // Figure out what the last time every script was run.
+    // If a script has not changed since the last time it was run,
+    // Don't run it.
+    var systemScriptCache = await AfterInstallCache.load(cache.rootDir);
+    var localScriptCache = p.equals(cache.rootDir, cachePath)
+        ? systemScriptCache
+        : await AfterInstallCache.load(cachePath);
+    var mergedCache = systemScriptCache.merge(localScriptCache);
+
+    // Sort so that the root's scripts run last.
+    var changedPackages = result.changedPackages.toList();
+    changedPackages.sort((a, b) => a == root.name ? 1 : a.compareTo(b));
+
+    // Run scripts in dependencies.
+    for (var package in changedPackages) {
+      var pkg = package == root.name
+          ? root
+          : cache.load(result.packages
+              .firstWhere((a) => a.name == package, orElse: () => null));
+
+      for (var script in pkg.pubspec.afterInstall) {
+        var absolutePath = p.normalize(p.absolute(p.join(pkg.dir, script)));
+        var scriptFile = new File(absolutePath);
+
+        // Don't run the script against there are new changes.
+        if (!await mergedCache.isOutdated(absolutePath)) continue;
+
+        if (!await scriptFile.exists()) {
+          if (pkg == root)
+            log.warning(log.yellow(
+                '"after_install" script "$absolutePath" does not exist.'));
+        } else {
+          // Never trust any third-party scripts. Show a prompt before running scripts.
+          //
+          // The user can opt to print the contents of the script.
+          YesNoPrintResponse response = YesNoPrintResponse.yes;
+
+          if (pkg != root) {
+            var message =
+                '\npackage:$package wants to run the script "$absolutePath".';
+
+            do {
+              response = await confirmOrPrint(message);
+
+              if (response == YesNoPrintResponse.no) {
+                log.message('User declined to run script.');
+                continue;
+              }
+
+              if (response == YesNoPrintResponse.print) {
+                log.message(log.gray('Contents of "$absolutePath":'));
+                log.message(log.gray(await scriptFile.readAsString()));
+              }
+            } while (response == YesNoPrintResponse.print);
+          }
+
+          if (response != YesNoPrintResponse.yes) continue;
+
+          // We need to change into the package's directory to run its scripts.
+          var currentDir = Directory.current;
+          Directory.current = pkg.dir;
```

Comment:
We shouldn't need to do this (and its probably not what many scripts would want, they may want to look at the current package and would expect that to be the current directory).

We definitely _dont_ want to run inside the package cache like this would do, as touching anything in there is forbidden.

Reply:
Sounds good, more than likely could be removed